misskey-websocket/src/channel.rs

```diff
@@ -211,8 +212,15 @@ where
     }
 }
 
-pub async fn connect_websocket(url: Url) -> Result<(WebSocketSender, WebSocketReceiver)> {
-    let (ws, _) = connect_async(url).await?;
+pub async fn connect_websocket(
+    uri: http::Uri,
+    additional_headers: HeaderMap,
+) -> Result<(WebSocketSender, WebSocketReceiver)> {
+    let request = uri.into_client_request()?;
+    let (mut parts, _) = request.into_parts();
+    parts.headers.extend(additional_headers);
+    let request = http::Request::from_parts(parts, ());
```

Comment:
`headers_mut` can be used instead
https://docs.rs/http/0.2.8/http/request/struct.Request.html#method.headers_mut

```suggestion
    let mut request = uri.into_client_request()?;
    request.headers_mut().extend(additional_headers);
```